File: Recursion_Array/reverse.cpp
#include<iostream>
#include<vector>
using namespace std;

void input(vector<int>& arr)
{
    for(int i = 0; i < arr.size(); i++)
    {
        cin >> arr[i];
    }
}

void revrse(vector<int>& arr, int idx)
{
    if(idx == -1)
    {
        return;
    }
    cout << arr[idx] << endl;
    revrse(arr, idx - 1);
}

int main()
{
    int n;
    cin >> n;
    vector<int> arr(n);
    input(arr);
    revrse(arr, n - 1);
    return 0;
}

File: Recursion_Array/reverse.java
import java.util.Scanner;
public class reverse
{
    public static Scanner scn = new Scanner(System.in);
    public static void reverse(int[] arr, int idx)
    {
        if(idx == -1)
        {
            return;
        }
        System.out.println(arr[idx]);
        reverse(arr, idx - 1);
    }

    public static void input(int[] arr)
    {
        for(int i = 0; i < arr.length; i++)
        {
            arr[i] = scn.nextInt();
        }
    }

    public static void main(String[] args)
    {
        int n = scn.nextInt();
        int[] arr = new int[n];
        input(arr);
        reverse(arr, n - 1);
    }
}